model2vec/train/classifier.py

```diff
@@ -186,45 +216,82 @@ def fit(
 
         self.load_state_dict(state_dict)
         self.eval()
-
         return self
 
-    def _initialize(self, y: list[str]) -> None:
-        """Sets the out dimensionality, the classes and initializes the head."""
-        classes = sorted(set(y))
-        self.classes_ = classes
-
-        if len(self.classes) != self.out_dim:
-            self.out_dim = len(self.classes)
+    def _initialize(self, y: list[str] | list[list[str]], multilabel: bool = False) -> None:
+        """
+        Sets the output dimensionality, the classes, and initializes the head.
 
+        :param y: The labels.
+        :param multilabel: Whether the task is multilabel.
+        """
+        self.multilabel = multilabel
+        if multilabel:
+            classes = sorted({label for sublist in y for label in sublist})
+        else:
+            classes = sorted(set(cast(list[str], y)))
+        self.classes_ = classes
+        self.out_dim = len(self.classes_)  # Update output dimension
         self.head = self.construct_head()
         self.embeddings = nn.Embedding.from_pretrained(self.vectors.clone(), freeze=False, padding_idx=self.pad_id)
         self.w = self.construct_weights()
         self.train()
 
-    def _prepare_dataset(self, X: list[str], y: list[str], max_length: int = 512) -> TextDataset:
-        """Prepare a dataset."""
+    def _prepare_dataset(
+        self, X: list[str], y: list[str] | list[list[str]], max_length: int = 512, multilabel: bool = False
+    ) -> TextDataset:
+        """
+        Prepare a dataset. For multilabel classification, each target is converted into a multi-hot vector.
+
+        :param X: The texts.
+        :param y: The labels.
+        :param max_length: The maximum length of the input.
+        :param multilabel: Whether the task is multilabel.
+        :return: A TextDataset.
+        """
         # This is a speed optimization.
         # assumes a mean token length of 10, which is really high, so safe.
         truncate_length = max_length * 10
         X = [x[:truncate_length] for x in X]
         tokenized: list[list[int]] = [
             encoding.ids[:max_length] for encoding in self.tokenizer.encode_batch_fast(X, add_special_tokens=False)
         ]
-        labels_tensor = torch.Tensor([self.classes.index(label) for label in y]).long()
-
+        if multilabel:
+            num_classes = len(self.classes)
+            label_list = []
+            for sample_labels in y:
+                multi_hot = torch.zeros(num_classes, dtype=torch.float)
```

Comment:
isn't this float by default?

Reply:
I also think it might be faster to just construct the whole tensor at once. So `multi_hot` = `torch.zeros(len(y), num_classes)`, and then you simply assign rows.